<?xml version="1.0"?>
<robot name="nonhol" 
	xmlns:xacro="http://www.ros.org/wiki/xacro">


	<!-- Included files -->
	<xacro:include filename="$(find node_counting_alg)/urdf/components.gazebo"/>
	<xacro:include filename="$(find node_counting_alg)/urdf/materials.xacro" />

	<!-- _______________________________________________________ -->

	<!-- Create robot's base_link -->
	<xacro:base_link name="base_link" size_x = "0.4" size_y = "0.2" size_z = "0.1" />

	<!-- Create wheels -->
	<xacro:wheel_link name="link_wheel1" />
	<xacro:wheel_joint name="joint_wheel1" child="link_wheel1" origin_xyz="0.15 0.15 0"/>

	<xacro:wheel_link name="link_wheel2" />
	<xacro:wheel_joint name="joint_wheel2" child="link_wheel2" origin_xyz="-0.15 0.15 0"/>

	<xacro:wheel_link name="link_wheel3" />
	<xacro:wheel_joint name="joint_wheel3" child="link_wheel3" origin_xyz="0.15 -0.15 0"/>

	<xacro:wheel_link name="link_wheel4" />
	<xacro:wheel_joint name="joint_wheel4" child="link_wheel4" origin_xyz="-0.15 -0.15 0"/>

	<!-- Create camera -->
	<xacro:sensor_link name="camera_link" color = "red"/>
	<xacro:sensor_joint name="camera_joint" child="camera_link" origin_xyz="-0.12 0 0.2" />

	<!-- Create laser -->
	<xacro:sensor_link name="laser_link" color = "blue"/>
	<xacro:sensor_joint name="laser_joint" child="laser_link" origin_xyz="0.12 0 0.2" />

	<!-- _______________________________________________________ -->

	<!-- Skid steer drive controller plugin -->
	<!-- (Here and not in the .gazebo because it needs radius_wheel param)  -->
	<gazebo>
		<plugin name="skid_steer_drive_controller" filename="libgazebo_ros_skid_steer_drive.so">
			<updateRate>	 100.0		  </updateRate>
			<leftFrontJoint> joint_wheel1 </leftFrontJoint>
			<rightFrontJoint>joint_wheel3 </rightFrontJoint>
			<leftRearJoint>	 joint_wheel2 </leftRearJoint>
			<rightRearJoint> joint_wheel4 </rightRearJoint>
			<wheelDiameter>	 0.1		  </wheelDiameter>
			<robotBaseFrame> base_link	  </robotBaseFrame>
			<torque>		 2			  </torque>
			<topicName>		 cmd_vel	  </topicName>
			<broadcastTF>	 1			  </broadcastTF>
		</plugin>
	</gazebo>

	<!-- Camera plugin -->
	<xacro:camera_plugin name="camera1" ref_link = "camera_link"/>

	<!-- Laser plugin -->
	<xacro:laser_plugin name="laser1" ref_link = "laser_link"/>

</robot>
